### ISSUE TITLE
need support asyncio

### ISSUE THREAD

**Issue by ArtemIsmagilov** [python-caldav/caldav#342]
I believe that the library needs asynchronous support - this will significantly improve the performance and efficiency of the library

**Comment by tobixen** [comment]
I agree, but I probably have no capacity to follow up this year.  Design ideas and pull-requests are welcome.   I'm in for anyhting that doesn't break backward-compatibility.

**Comment by tobixen** [comment]
https://github.com/python-caldav/caldav/pull/610

It's been a long journey, experimenting and looking into different ways of doing this, but finally I've settled for a "sans-io"-approach to it.

**Comment by tobixen** [comment]
Pull request #610 has been merged to master, so the master version (will be released as 3.0.0-a1 tomorrow probably) is now async.
